packages/api-fetch/src/middlewares/preloading.js

```diff
@@ -45,8 +45,10 @@ function createPreloadingMiddleware( preloadedData ) {
 			const method = options.method || 'GET';
 			const path = getStablePath( options.path );
 
-			if ( parse && 'GET' === method && cache[ path ] ) {
-				return Promise.resolve( cache[ path ].body );
+			if ( 'GET' === method && cache[ path ] ) {
+				return Promise.resolve(
+					parse ? cache[ path ].body : { json: cache[ path ] }
```

Comment:
Can we create a `new Headers` instance with the headers data? That way we don't have to do the `response.headers.get` introspection?

Reply:
What's the benefit of that?

Reply:
I think this can be a pretty obnoxious pattern to follow.

```
response.headers.get
 			? response.headers.get( 'link' )
 			: response.headers.Link
```

If we make it a `new Headers( cache[ path ].headers )`, then the consumer can always use `response.headers.get( 'Link' )`.

I think it's also safer in the sense that the receiver of an `apiFetch` response isn't necessarily in control of whether or not that request has been preloaded. So if they are always using `.get()` now, but some day someone else or Gutenberg preloads the same query they were making, their code will now break.

Reply:
Ahh, yes.

Updated. I also made `json` return a promise.